rhombus/private/array.rkt

```diff
@@ -305,6 +306,21 @@
   (check-array who v)
   (vector->treelist v))
 
+(define-sequence-syntax Array.to_sequence/optimize
+  (lambda () #'Array.to_sequence)
+  (lambda (stx)
+    (syntax-parse stx
+      [[(id) (_ arr-expr)]
+       #'[(id) (in-vector (let ([arr arr-expr])
+                            (check-array 'Array.to_sequence arr)
+                            arr))]]
+      [_ #f])))
```

Comment:
Does this generate the same code as before, or does it add an extra layer of checking? How about in unsafe mode? I wonder whether this should instead be handled by rewriting the message from a existing check (although that might require updating `in-list` to compare as being in the `racket/primitive` realm instead of `racket`).

Reply:
It doesn’t generate the same code.  Rewriting requires some cooperation at Racket side, changing relevant checks in `racket/private/for` (to be more consistent with `in-treelist`, for that matter).

Reply:
I pushed a change `racket/private/for`.